enterprise/server/remote_execution/containers/firecracker/firecracker.go

```diff
@@ -693,25 +693,27 @@ func MergeDiffSnapshot(ctx context.Context, baseSnapshotPath string, baseSnapsho
 				if offset >= regionEnd {
 					break
 				}
-				n, err := gin.ReadAt(buf, offset)
-				if err != nil && err != io.EOF {
-					return err
+				n, readErr := gin.ReadAt(buf, offset)
+				if readErr != nil && readErr != io.EOF {
+					return readErr
 				}
 
 				if _, err := out.WriteAt(buf[:n], offset); err != nil {
 					return err
 				}
 
-				// After the store has been updated, unmap the chunk to save memory
+				// If we've finished processing a chunk, unmap it to save memory
 				// usage on the executor
-				if baseSnapshotStore != nil {
+				nextOffsetInNextChunk := (offset%cowChunkSizeBytes())+int64(n) >= cowChunkSizeBytes()
```

Comment:
uber nit: should we call `baseSnapshotStore.ChunkSize()` so we're using the actual chunk size of the file? (in case we wind up changing the chunk size later)